opm/simulators/utils/readDeck.cpp

```diff
@@ -241,28 +240,66 @@ void readDeck(int rank, std::string& deckFilename, std::unique_ptr<Opm::Deck>& d
         if (!eclipseState)
             eclipseState = std::make_unique<Opm::ParallelEclipseState>();
     }
+#endif
+
+    if (*errorGuard) { // errors encountered
+        parseSuccess = 0;
+    }
+
+    // print errors and warnings!
+    errorGuard->dump();
+    errorGuard->clear();
 
     auto comm = Dune::MPIHelper::getCollectiveCommunication();
-    parseSuccess = comm.max(parseSuccess);
+    parseSuccess = comm.min(parseSuccess);
+
     if (!parseSuccess)
     {
-        if (*errorGuard) {
-            errorGuard->dump();
-            errorGuard->clear();
+        if (rank == 0)
+        {
+            OpmLog::error(std::string("Unrecoverable errors were encountered while loading input: ")+failureMessage);
         }
-        MPI_Abort(MPI_COMM_WORLD, EXIT_FAILURE);
+#if HAVE_MPI
+        MPI_Finalize();
+#endif
+        exit(1);
```

Comment:
This should be
```C++
std::exit(EXIT_FAILURE);
```
too&mdash;preferably with a `std::` prefix.

Reply:
done